purchase_order_bl_date/models/stock_move.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2023 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import models, api, fields
+import logging
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    @api.multi
+    def _get_price_unit(self):
```

Comment:
Is it possible to override `_get_conversion_rate` and pass `bl_date`? 
we would like to avoid a complete override.

Another my concern is that when you `button_confirm` in purhase_order with `bl_date`,  price_unit of the stock.move record that is created is not affected by the bl_date.
Because if I am not mistaken, the method `_get_stock_move_price_unit` is used when the stock.move record is first created and this `_get_price_unit` is not used. 

Can you check that is it a problem that stock.move is calculated and created at today's rate even if the `bl_date` is set?

Reply:
@kanda999 
The `_get_conversion_rate` is not called from `stock` module when picking is created from purchase or when they create stock valuation. So, I have no idea how to `_get_conversion_rate` and pass `bl_date`. I also wanted to avoid complete override `_get_price_unit` but this function only returns the converted price_unit and we can't convert again. And I see odoo override like this https://github.com/odoo/odoo/blob/0e56e4a7bd6de42d729441a53995ddd459d5e633/addons/purchase_stock/models/stock.py#L37.

I also knew `_get_stock_move_price_unit` is called when we click `button_confirm` in `purchase_order`. But when we calculate stock valuation for journal entry, it works with _get_price_unit(). But I agree we should use _get_stock_move_price_unit for design consistent.

Reply:
Can we do like this:

```python
    @api.multi
    def _get_price_unit(self):
        bl_date = self.picking_id.bl_date
        if bl_date:
            self = self.with_context(bl_date=bl_date)
        return super()._get_price_unit()
```

And then (in `res.currency`):

```python
    def _convert(self, from_amount, to_currency, company, date, round=True):
        bl_date = self._context.get("bl_date")
        date = bl_date or date
        return _convert(from_amount=fro_amount, to_currency=to_currency, company=company, date=date, round=round)
```